lib/grant-types/implicit-grant-type.ts

```diff
@@ -0,0 +1,63 @@
+import { AbstractGrantType } from '.';
+import { InvalidArgumentError } from '../errors';
+import { Client, Token, User } from '../interfaces';
+import { Request } from '../request';
+
+export class ImplicitGrantType extends AbstractGrantType {
+  scope: string;
+  user: User;
+  constructor(options: any = {}) {
+    super(options);
+
+    if (!options.model) {
+      throw new InvalidArgumentError('Missing parameter: `model`');
+    }
+
+    if (!options.model.saveToken) {
+      throw new InvalidArgumentError(
+        'Invalid argument: model does not implement `saveToken()`',
+      );
+    }
+
+    if (!options.user) {
+      throw new InvalidArgumentError('Missing parameter: `user`');
+    }
+
+    this.scope = options.scope;
+    this.user = options.user;
+  }
+
+  /**
+   * Handle implicit token grant.
+   */
+
+  async handle(request: Request, client: Client) {
+    if (!request) {
+      throw new InvalidArgumentError('Missing parameter: `request`');
+    }
+
+    if (!client) {
+      throw new InvalidArgumentError('Missing parameter: `client`');
+    }
+
+    return this.saveToken(this.user, client, this.scope);
+  }
+
+  /**
+   * Save token.
+   */
+
+  async saveToken(user: User, client: Client, scope: string) {
+    const validatedScope = await this.validateScope(user, client, scope);
+    const accessToken = await this.generateAccessToken(client, user, scope);
```

Comment:
The might resolve faster in parallel.

Reply:
Like I commented above

In this case it is better to not run them all in parallel just because it will process faster as it is security relevant and we need to validate the scope before we do the next steps